doc/developers/contributing.rst

```diff
@@ -734,10 +736,19 @@ In general have the following in mind:
     5. Specify ``dataframe`` when "frame-like" features are being used, such
        as the column names.
     6. When specifying the data type of a list, use ``of`` as a delimiter:
-       ``list of int``.
+       ``list of int``. When the parameter supports arrays giving details about
+       the shape and/or data type and a list of such arrays, you can use one
+       of:
+
+      * ``array-like of shape (n_samples,) or list thereof``;
+      * ``array-like of shape (n_samples,) or list of these``;
+      * ``array-like of shape (n_samples,) or list of such arrays``.
+
     7. When specifying the dtype of an ndarray, use e.g. ``dtype=np.int32``
        after defining the shape:
-       ``ndarray of shape (n_samples,), dtype=np.int32``.
+       ``ndarray of shape (n_samples,), dtype=np.int32``. You can specify
+       multiple dtype as a set:
+       ``array-like of shape (n_samples,), dtype={int, float}``.
```

Comment:
I would rather not recommend to use `dtype=int` in documentation. Either use a specific precision level (`np.int32`) or "integer dtype" or floating point dtype" if you want to mention arbitrary precision. Or just "numeric dtype" if both integer or floating point dtypes are accepted (but maybe dtype can be completely omitted in this case.

Reply:
With numpy, passing `dtype=int` and `dtype=float` will result in `np.int64` and `np.float64` as actual dtypes but:

- this is implicit;
- this might be platform specific and could lead to subtle bugs (though I am not sure it's still the case with recent numpy versions and Python 3).

Reply:
> this might be platform specific and could lead to subtle bugs (though I am not sure it's still the case with recent numpy versions and Python 3).

I think this is still the case with the integer (or at least last week during a course :))

Reply:
I made the correction avoiding using `int` and `float` and advise for using `integer`, `floating` and `numeric`.
I added some explanation in the glossary as well.